prost-build/src/code_generator.rs

```diff
@@ -246,14 +293,8 @@ impl<'a> CodeGenerator<'a> {
             }
             self.path.pop();
 
-            for (idx, oneof) in message.oneof_decl.into_iter().enumerate() {
-                let idx = idx as i32;
-                // optional fields create a synthetic oneof that we want to skip
-                let fields = match oneof_fields.remove(&idx) {
-                    Some(fields) => fields,
-                    None => continue,
-                };
-                self.append_oneof(&fq_message_name, oneof, idx, fields);
```

Comment:
This was done with a moving iterator and moving the field data into `append_oneof`, but there was no reason to move the data because code generation only needs references. And I need the same data for the builder pass which will be added later.

Reply:
I see, the code just moved. That de-duplucation is nice.